tkp/accessors/fitsimage.py

```diff
@@ -124,23 +124,34 @@ def parse_frequency(self):
         try:
             header = self.header
             if header['TELESCOP'] in ('LOFAR', 'AARTFAAC'):
-                freq_eff = header['RESTFRQ']
-                if 'RESTBW' in header:
-                    freq_bw = header['RESTBW']
+                if 'RESTFRQ' in header:
+                    freq_eff = header['RESTFRQ']
+                    if 'RESTBW' in header:
+                        freq_bw = header['RESTBW']
 
-                else:
-                    logger.warning("bandwidth header missing in image {},"
+                    else:
+                        logger.warning("bandwidth header missing in image {},"
                                    " setting to 1 MHz".format(self.url))
-                    freq_bw = 1e6
+                        freq_bw = 1e6
+                else:
+                    if ('CTYPE3' in header) and (header['CTYPE3'] in ('FREQ', 'VOPT')):
+                        freq_eff = header['CRVAL3']
+                        freq_bw = header['CDELT3']
+                    elif ('CTYPE4' in header) and (header['CTYPE4'] in ('FREQ', 'VOPT')):
+                        freq_eff = header['CRVAL4']
+                        freq_bw = header['CDELT4']
+                    else:
+                        freq_eff = header['RESTFREQ']
+                        freq_bw = 0.0
             else:
-                if ('ctype3' in header) and (header['ctype3'] in ('FREQ', 'VOPT')):
-                    freq_eff = header['crval3']
-                    freq_bw = header['cdelt3']
-                elif ('ctype4' in header) and (header['ctype4'] in ('FREQ', 'VOPT')):
-                    freq_eff = header['crval4']
-                    freq_bw = header['cdelt4']
+                if ('CTYPE3' in header) and (header['CTYPE3'] in ('FREQ', 'VOPT')):
+                    freq_eff = header['CRVAL3']
+                    freq_bw = header['CDELT3']
+                elif ('CTYPE4' in header) and (header['CTYPE4'] in ('FREQ', 'VOPT')):
+                    freq_eff = header['CRVAL4']
+                    freq_bw = header['CDELT4']
                 else:
-                    freq_eff = header['restfreq']
+                    freq_eff = header['RESTFREQ']
```

Comment:
Out of interest was it a conscious choice for this too be `RESTFREQ` and the above LOFAR one to be `RESTFRQ`?

Reply:
This is legacy code and really dependent on what you use to make the images. So best to keep for different imagers.